commons/es_client.py

```diff
@@ -51,15 +55,37 @@ def index_exists(self, es_index_name, print_error=True):
                 logger.error(err)
             return False
 
-    def delete_index(self, index_name):
+    def delete_index(self, project_id):
         """Delete the whole index"""
-        es_index_name = self.format_index_name(index_name)
+        es_index_name = self.format_index_name(project_id)
         if self.index_exists(es_index_name):
             try:
                 self.es_client.indices.delete(index=es_index_name + "*")
-                # TO DO delete template
-                # TO DO delete policy
+                delete_template_response = requests.delete(
+                    f"{self.host}/_index_template/{self.get_template_name(es_index_name)}",
+                    headers={'Content-type': 'application/json', 'Accept': 'text/plain'}
+                ).__dict__
+                delete_policy_response = requests.delete(
+                    f"{self.host}/_ilm/policy/{self.get_policy_name(es_index_name)}",
+                    headers={'Content-type': 'application/json', 'Accept': 'text/plain'}
+                ).__dict__
                 logger.info("ES Url %s", utils.remove_credentials_from_url(self.host))
+                if delete_template_response["status_code"] == 200:
```

Comment:
the logic of error logging for template and policy is too similar, let's create a function for it

Reply:
done 
![image](https://user-images.githubusercontent.com/40484210/117001389-f654a800-acea-11eb-86b1-a75331d9aa4a.png)